x-pack/plugin/identity-provider/src/main/java/org/elasticsearch/xpack/idp/action/SamlGenerateMetadataAction.java

```diff
@@ -0,0 +1,18 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+package org.elasticsearch.xpack.idp.action;
+
+import org.elasticsearch.action.ActionType;
+
+public class SamlGenerateMetadataAction extends ActionType<SamlGenerateMetadataResponse> {
```

Comment:
I don't think `Generate` is needed in these names. `SamlMetadataAction` would be sufficient.